common4j/src/main/com/microsoft/identity/common/java/cache/MicrosoftFamilyOAuth2TokenCache.java

```diff
@@ -212,13 +212,13 @@ public ICacheRecord loadByFamilyId(@Nullable final String clientId,
                         + " profiles for this account"
         );
 
-        // Ignore the first element, as it will contain the same result as loadByFamilyId()....
-        accountsInOtherTenants.remove(0);
-
         if (!accountsInOtherTenants.isEmpty()) {
             // We need the IdToken of each of these accounts... we can reuse the RT, since it is
             // an FRT...
 
+            // Ignore the first element, as it will contain the same result as loadByFamilyId()....
+            accountsInOtherTenants.remove(0);
```

Comment:
Relatively minor, but maybe something to think about: if this result in an IndexOutOfBounds exception, that means that this is backed by an ArrayList, so that a remove operation will result in copying the entire array... It would be nice to avoid that, if we can, since we're already copying it once.